datafusion/sql/src/unparser/dialect.rs

```diff
@@ -168,6 +173,17 @@ pub enum DateFieldExtractStyle {
     Strftime,
 }
 
+/// `CharacterLengthStyle` to use for unparsing
+///
+/// Different DBMSs uses different names for function calculating the number of characters in the string
+/// `Length` style uses length(x)
+/// `SQLStandard` style uses character_length(x)
+#[derive(Clone, Copy, PartialEq)]
+pub enum CharacterLengthStyle {
```

Comment:
Introducing additional `CharacterLengthStyle` abstraction might be an overkill, other changes are LGTM